### PR TITLE
BUG: fix+test PA+all-NaT TDA

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

In conjunction with one more branch coming up, we're going to simplify the tar out of Series arithmetic ops.

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -157,13 +157,22 @@ def maybe_upcast_for_op(obj):
...
+    elif isinstance(obj, np.timedelta64):
+        if isna(obj):
+            # wrapping timedelta64("NaT") in Timedelta returns NaT,
+            #  which would incorrectly be treated as a datetime-NaT, so
+            #  we broadcast and wrap in a Series
+            right = np.broadcast_to(obj, shape)
+
+            # Note: we use Series instead of TimedeltaIndex to avoid having
+            #  to worry about catching NullFrequencyError.
+            return pd.Series(right)
```
Comment (jreback) [inline pandas/core/ops/__init__.py]: rather than this can you make a safe method that deals with null frequency, this is a bit odd

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py]: This basically is the safe method to deal with null frequency.  I think there will be room to simplify it after we merge dispatch_to_index_op into dispatch_to_extension_op

(I agree that I'd prefer these methods not need to know about Series)

Comment (jreback) [reply pandas/core/ops/__init__.py]: ok fair enough, maybe make an issue about this? (or just add to your list) :->

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**